### PR TITLE
Integrate Mdox to the Project

### PR DESCRIPTION
**Pull request by AshwinSriram11**
This PR will integrate [mdox](https://github.com/bwplotka/mdox?tab=readme-ov-file#link-localization) to this project for formatting and link-checking all the docs. We will add CI checks for link-checking.

With this PR, a contributor can check the formatting locally using mdox. `make docs` is used for `mdox fmt` and `make check docs` is used for `mdox fmt --check` .

This PR will solve the issue #91. Currently, it is in draft phase.

### CODE REVIEW

**Review by ArthurSens — COMMENTED** [review]
Good start! I think there are some steps that are super easy to miss, given the complexity of our Makefile in the main repo.

Steps missing:
* File `scripts/tools.go`.  In the [main repo](https://github.com/prometheus-operator/prometheus-operator/blob/main/scripts/tools.go), we use this file just to aggregate binaries that we use in the Makefile.

* Then we have a [step called `$(TOOLING)`, where we download all the binaries](https://github.com/prometheus-operator/prometheus-operator/blob/a7a1ad63492f1de3b09b2c960c7159b16ce00e84/Makefile#L417-L422). In the tooling step, we also [declare all binaries that need to be installed](https://github.com/prometheus-operator/prometheus-operator/blob/a7a1ad63492f1de3b09b2c960c7159b16ce00e84/Makefile#L48).

* [In our `docs` and `check-docs` steps](https://github.com/prometheus-operator/prometheus-operator/blob/a7a1ad63492f1de3b09b2c960c7159b16ce00e84/Makefile#L327-L335), we say that the `$(MDOX_BINARY)` is a dependency.

Then, when we call `docs` or `check-docs`, the flow is:
```mermaid
graph TD;
docs["`make docs`"]
mdoxbinary["$(MDOX_BINARY)"]
tooling["$(TOOLING)"]
tools["Download binaries from scripts/tools.go"]
run["Finally execute mdox"]
docs-->mdoxbinary;
mdoxbinary-->tooling;
tooling-->tools;
tools-->run;

```

File: content/docs/getting-started/installation.md [hunk 1]
```diff
@@ -1,14 +1,14 @@
 ---
```
[the remark this answers is not in the document]
Comment (AshwinSriram11) [reply content/docs/getting-started/installation.md:1]: Do you mean that I should make changes in `make docs` and `make check-docs` and make a yaml config in .github for CI with the wrong config?

[the remark this answers is not in the document]
Comment (ArthurSens) [reply content/docs/getting-started/installation.md:1]: No, just that the initial commit shouldn't make any changes to the markdown files. Just a failing CI workflow and Makefile changes to download mdox

**Comment by AshwinSriram11** [comment]
This PR is not needed as described in issue [here](https://github.com/prometheus-operator/website/issues/91#issuecomment-2293306110)

**Pull request closed without merging**